frontend/src/components/modals/CreateApiKeyModal/index.tsx

```diff
@@ -0,0 +1,67 @@
+import {useForm} from "@mantine/form";
+import {GenericModalProps, CreateApiKeyRequest, ApiKey} from "../../../types.ts";
+import {Modal} from "../../common/Modal";
+import {Button} from "@mantine/core";
+import {useFormErrorResponseHandler} from "../../../hooks/useFormErrorResponseHandler.tsx";
+import {t, Trans} from "@lingui/macro";
+import {useCreateApiKey} from "../../../mutations/useCreateApiKey.ts";
+import {showSuccess, showError} from "../../../utilites/notifications.tsx";
+import {ApiKeyForm} from "../../forms/ApiKeyForm/index.tsx";
+import {apiKeysClient} from "../../../api/api-keys.client.ts";
+import {useMutation, useQueryClient} from "@tanstack/react-query";
+import {GET_API_KEYS_QUERY_KEY} from "../../../queries/useGetApiKeys.ts";
+
+interface CreateApiKeyModalProps extends GenericModalProps {
+    onCompleted: (apiKey: ApiKey) => void;
+}
+
+// TODO: translations
+export const CreateApiKeyModal = ({onClose, onCompleted}: CreateApiKeyModalProps) => {
+    const createMutation = useCreateApiKey();
+    const queryClient = useQueryClient();
+    const formErrorHandler = useFormErrorResponseHandler();
+
+    const form = useForm<CreateApiKeyRequest>({
+        initialValues: {
+            token_name: '',
+            abilities: [],
+            expires_at: '',
+        },
+    });
+
+    const handleCreate = useMutation({
+        mutationFn: (apiKeyRequest: CreateApiKeyRequest) => apiKeysClient.create(apiKeyRequest),
+
+        onSuccess: (response: NewApiKey) => {
```

Comment:
`NewApiKey ` doesn't exist

Reply:
The interface [does exist](https://github.com/HiEventsDev/Hi.Events/pull/350/files/334528cce77c01cbf1ae77b47629b1874e53e916#diff-19743365133d63884e16c2ed111a9076bb50a34785a3e624bd4abf49d49f814cR698), but you're right, I didn't import it in this file and I didn't specify it as the result of a Create call in the api-keys.client.ts file.

Thanks!